### PR TITLE
Set the default return type to 'application/json'

### LINKED ISSUE #1850: Fix JSON content detection
While working on #1616 i broke the [`_get_response_content()`](https://github.com/privacyidea/privacyidea/blob/a128999d61f7d6fed114ed371528ab7d766be4d8/privacyidea/lib/eventhandler/base.py#L303) function in 3fdac34.
In case a response does not contain any JSON data, this would result in an error.
This would happen if a post-event was bound to an endpoint like `/validate/radiuscheck`.

### PR DESCRIPTION
**Pull request by plettich**
Since we mostly return JSON data, it makes sense to the set it as the
default return type in out response class.
Also added a `send_file()` utility function which adjusts content type and
header accordingly.
Other mimetypes are set explicitly.

Fixes #1850

### CODE REVIEW

File: privacyidea/api/lib/utils.py [hunk 1]
```diff
@@ -148,6 +148,12 @@ def send_error(errstring, rid=1, context=None, error_code=-311, details=None):
     return ret
 
 
+def send_file(output, filename):
+    content_type = "application/force-download"
```
Comment (fredreichbier) [inline privacyidea/api/lib/utils.py]: I'm not sure about the ``application/force-download`` mimetype here. I didn't know it, so I googled it, and [it seems like](https://stackoverflow.com/a/10616753) only shouldn't really use it, but instead set the actual content mimetype. I'm wondering -- should we do that instead? What do you think?

File: privacyidea/api/lib/utils.py [hunk 2]
```diff
@@ -148,6 +148,12 @@ def send_error(errstring, rid=1, context=None, error_code=-311, details=None):
     return ret
 
 
+def send_file(output, filename):
```
Comment (fredreichbier) [inline privacyidea/api/lib/utils.py]: For the final PR, could you add a small docstring?

File: tests/test_api_audit.py [hunk 3]
```diff
@@ -25,6 +28,38 @@ def test_00_get_audit(self):
             self.assertTrue(json_response.get("result").get("value").get(
                 "current") == 1, res)
 
+    def test_01_get_audit_csv(self):
```
Comment (fredreichbier) [inline tests/test_api_audit.py]: nice testcase!

File: privacyidea/api/lib/utils.py (hunk 2 above)
Comment (plettich) [reply privacyidea/api/lib/utils.py]: Done in ba5ab0b

File: privacyidea/api/lib/utils.py (hunk 1 above)
Comment (plettich) [reply privacyidea/api/lib/utils.py]: Fixed in ba5ab0b

File: privacyidea/webui/login.py [hunk 4]
```diff
@@ -58,7 +58,8 @@ def single_page_application():
 
     if current_app.config.get("PI_UI_DEACTIVATED"):
         # Do not provide the UI
-        return render_template("deactivated.html")
+        return current_app.response_class(render_template("deactivated.html"),
```
Comment (cornelinux) [inline privacyidea/webui/login.py]: We often use the ``current_app.respsonse_class(render_template...)`` which return an html mimetype. Would it be nice to have a wrapper funktion for this? 
(e.g. the same in webui/certificate.py)

File: privacyidea/webui/certificate.py [hunk 5]
```diff
@@ -98,14 +100,14 @@ def cert_enroll():
...
-
+    render_context = {'instance': instance,
+                      'backendUrl': backend_url,
+                      'username': "{0!s}@{1!s}".format(request.PI_username,
+                                                       request.PI_realm),
+                      'role': request.PI_role,
+                      'serial': serial,
+                      'certificate': certificate,
+                      'cert_pem': cert_pem }
+    return current_app.response_class(render_template("token_enrolled.html",
```
Comment (cornelinux) [inline privacyidea/webui/certificate.py]: see below

File: privacyidea/webui/certificate.py [hunk 6]
```diff
@@ -58,9 +58,11 @@ def cert_form():
     backend_url = ""
     authtoken = request.all_data.get("authtoken")
     ca = request.all_data.get("ca")
-    return render_template("cert_request_form.html", instance=instance,
-                           backendUrl=backend_url, ca=ca,
-                           authtoken=authtoken)
+    return current_app.response_class(render_template("cert_request_form.html",
```
Comment (cornelinux) [inline privacyidea/webui/certificate.py]: see below

File: privacyidea/webui/login.py (hunk 4 above)
Comment (plettich) [reply privacyidea/webui/login.py]: Fixed in 12f8421e

File: privacyidea/webui/certificate.py (hunk 5 above)
Comment (plettich) [reply privacyidea/webui/certificate.py]: see above...

File: privacyidea/webui/certificate.py (hunk 6 above)
Comment (plettich) [reply privacyidea/webui/certificate.py]: see above...

**Pull request merged**